backend/src/airtable-api.js

```diff
@@ -0,0 +1,147 @@
+      .eachPage(
+        (records, fetchNextPage) => {
+          allRecords = allRecords.concat(records);
+          fetchNextPage();
+        },
+        (err) => {
+          if (err) reject(err);
+          resolve(allRecords);
+        }
+      );
+  });
+
+  // Handle the promise we created above
+  let records = [];
+  try {
+    records = await out;
+  } catch (err) {
+    console.error('fetch airtable bills error', err);
+    throw err;
+  }
+
+  // Key campaign to assembly/senate bill id
+  const mapping = {};
+  records.forEach((record) => {
+    const senateId = record.get(BILL_SENATE_FIELD_ID);
+    const assemblyId = record.get(BILL_ASSEMBLY_ID_FIELD_ID);
+    const campaignId = record.get(BILL_CAMPAIGN_FIELD_ID);
+    if (!campaignId) return;
+
+    // Add campaign to the senate bill
+    if (senateId) {
+      if (senateId.startsWith('S')) {
+        if (!mapping[senateId]) {
+          mapping[senateId] = [];
+        }
+        mapping[senateId].push(campaignId);
+      } else console.log('Senate ID not properly formatted:', senateId);
+    }
+
+    // Add campaign to the assembly bill
+    if (assemblyId) {
+      if (assemblyId.startsWith('A')) {
+        if (!mapping[assemblyId]) {
+          mapping[assemblyId] = [];
+        }
+        mapping[assemblyId].push(campaignId);
+      } else console.log('Assembly ID not properly formatted:', assemblyId);
+    }
+  });
+
+  return mapping;
+};
+
+/** Fetches campaigns & their metadata */
+const sunriseCampaigns = async () => {
+  // NOTE: Airtable fetches aren't promises by default, wtf...
+  // Here we wrap the fetch in a promise to make it easier to work with
+
+  const out = new Promise((resolve, reject) => {
+    let allRecords = [];
+    base(CAMPAIGNS_TABLE_ID)
+      .select({
+        view: 'Grid view',
+        fields: [CAMPAIGN_LONG_NAME_FIELD_ID, CAMPAIGN_SHORT_NAME_FIELD_ID],
+      })
+      .eachPage(
+        (records, fetchNextPage) => {
+          allRecords = allRecords.concat(records);
+          fetchNextPage();
+        },
+        (err) => {
+          if (err) reject(err);
+          resolve(allRecords);
+        }
+      );
+  });
+
+  let records = [];
+  try {
+    records = await out;
+  } catch (err) {
+    console.error('fetch airtable campaigns error', err);
+    throw err;
+  }
+
+  const campaigns = {};
+  records.forEach((record) => {
+    const longName = record.get(CAMPAIGN_LONG_NAME_FIELD_ID);
+    const shortName = record.get(CAMPAIGN_SHORT_NAME_FIELD_ID);
+    const campaignId = record.id;
+    campaigns[campaignId] = {
+      long_name: longName,
+      short_name: shortName,
+      id: campaignId,
+    };
+  });
+
+  return campaigns;
+};
+
+export const fetchSunriseBills = async () => {
+  let bills = {};
+  let campaigns = {};
+  try {
+    bills = await sunriseBills();
+    campaigns = await sunriseCampaigns();
+  } catch (err) {
+    console.error('fetch airtable bills or campaigns error', err);
+  }
+  return { bills, campaigns };
```

Comment:
Not a fan of this return formatting, but there's no reason to break this into two endpoints.

Reply:
Looks good to me. Especially if you add a slight change to execute the calls concurrently:

```
[bills, campaigns] = await Promise.all([
      sunriseBills(),
      sunriseCampaigns()
    ]);
```

Reply:
Makes sense. Made the change, and re-requesting review.